doc/tutorial/argument_parser/index.md

```diff
@@ -0,0 +1,414 @@
+
+## Command line argument terminology
+
+Before we start, let's note down some terminology. Consider the following command line call
+```sh
+mycomputer$ ./program1 -i 4 --long-id 6 -f file1.txt
+```
+The binary `program1` is called with several command line **arguments**. We call every single input an **argument** but differentiate their prupose into **options**, **positional_options**, **flags** or simply a **value** corresponding to one of the former. In our command line call example, `-i` is an **option** with a short identifier (id) followed by its **value** `4`, `--long-id` is also an **option** with a long id followed by its **value** `6`,  `-f` is a **flag** which is never followed by a value, and **file1.txt** is a **positional_option** because it is an option identified by its position instead of an identifier.
+
+| Name                   | Purpose                                        | Example                 |
+|:-----------------------|:-----------------------------------------------|:------------------------|
+| **option**             | identify an argument by name (*id-value pair*) | `-i 5` or `--long-id 5` |
+| **flag**               | boolean on/off flag (*id*)                     | `-f`                    |
+| **positional option**  | identify an argument by position (*value*)     | `file1.txt`             |
+
+Have a look at the [POSIX conventions](https://www.math.uni-hamburg.de/doc/java/tutorial/essential/attributes/_posix.html) for command line arguments if you want a detailed description on the requirements for the above. (Note: in the linked article the following holds: value="argument", option="option", flag = "option that does not require arguments", positional option ="non-option").
+
+## A continuous example
+We will get to know the wide functionality of the argument parser by writing a little application and extending it step by step.  Let's say we have a tab separated file `data.tsv` with information on the Game of Thrones Seasons ([by wikipedia](https://en.wikipedia.org/wiki/List_of_Game_of_Thrones_episodes)):
+
+\include doc/tutorial/argument_parser/data.tsv
+
+We want to build an application that is able to read the file with or without a header line, select certain seasons and compute the average or meadian from the "Avg. U.S. viewers (millions)" of the selected seasons.
+
+# The SeqAn3 argument parser class
+
+Before we add any of the options, flags and positional options we will take a look at the seqan3::argument_parser class itself. It is constructed by giving a name and passing the parameters `argc` and `argv` from main. Note that no command line arguments have been parsed so far but we can now add more information to the parser.  After adding all desired information, the parsing is triggered by calling the **`parse()`** member function. Since it will throw in case any errors occur, we need to wrap it into a try-catch block. Here is a first working example:
```

Comment:
```suggestion
Before we add any of the options, flags and positional options we will take a look at the seqan3::argument_parser class itself. It is constructed by giving a name and passing the parameters `argc` and `argv` from main. Note that no command line arguments have been parsed so far but we can now add more information to the parser. After adding all desired information, the parsing is triggered by calling the **`parse()`** member function. Since it will throw in case any errors occur, we need to wrap it into a try-catch block. Here is a first working example:
```

Reply:
`[...] parser.__After [...]` to `[...] parser._After [...]` (double whitespace)